src/Spryker/Zed/AppKernel/Persistence/AppKernelRepository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace Spryker\Zed\AppKernel\Persistence;
+
+use Generated\Shared\Transfer\AppConfigCriteriaTransfer;
+use Spryker\Shared\Kernel\Transfer\TransferInterface;
+use Spryker\Shared\Log\LoggerTrait;
+use Spryker\Zed\AppKernel\Persistence\Exception\AppConfigNotFoundException;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+
+/**
+ * @method \Spryker\Zed\AppKernel\Persistence\AppKernelPersistenceFactory getFactory()
+ */
+class AppKernelRepository extends AbstractRepository implements AppKernelRepositoryInterface
+{
+    use LoggerTrait;
+
+    /**
+     * @param \Generated\Shared\Transfer\AppConfigCriteriaTransfer $appConfigCriteriaTransfer
+     * @param \Spryker\Shared\Kernel\Transfer\TransferInterface $transfer
+     *
+     * @throws \Spryker\Zed\AppKernel\Persistence\Exception\AppConfigNotFoundException
+     *
+     * @return \Spryker\Shared\Kernel\Transfer\TransferInterface
+     */
+    public function findAppConfigByCriteria(
+        AppConfigCriteriaTransfer $appConfigCriteriaTransfer,
+        TransferInterface $transfer
+    ): TransferInterface {
+        $appConfigEntity = $this->getFactory()
+            ->createAppConfigQuery()
+            ->findOneByTenantIdentifier($appConfigCriteriaTransfer->getTenantIdentifierOrFail());
+
+        if (!$appConfigEntity) {
+            $errorMessage = sprintf('Could not find an App configuration for the Tenant: %s', $appConfigCriteriaTransfer->getTenantIdentifierOrFail());
```

Comment:
as @k4emic mentioned before - we have decided not to have any dynamic log messages; all dynamic info should go into context or extra fields.

Reply:
Fixed